services/argentPortfolioService.ts

```diff
@@ -31,23 +36,67 @@ const fetchData = async <T>(endpoint: string): Promise<T> => {
   }
 };
 
+const fetchDataWithRetry = async <T>(
+  endpoint: string,
+  options?: {
+    maxRetries?: number;
+    initialDelay?: number;
+    maxDelay?: number;
+    backoffFactor?: number;
+    filterErrorRetryable?: (error: Error) => boolean;
+  }
+): Promise<T> => {
+  const maxRetries = options?.maxRetries ?? DEFAULT_MAX_RETRIES;
+  const initialDelay = options?.initialDelay ?? DEFAULT_INITIAL_DELAY;
+  const maxDelay = options?.maxDelay ?? DEFAULT_MAX_DELAY;
+  const backoffFactor = options?.backoffFactor ?? DEFAULT_BACKOFF_FACTOR;
+  const filterErrorRetryable = options?.filterErrorRetryable ?? (() => true);
+  let retries = 0;
+  let delay = initialDelay;
+  let lastError: Error = new Error('No request attempted');
+
+  while (retries <= maxRetries) {
+    try {
+      return await fetchData<T>(endpoint);
+    } catch (err) {
+      lastError = err instanceof Error ? err : new Error(String(err));
+
+      if (!filterErrorRetryable(lastError)) {
+        throw lastError;
+      }
+
+      if (retries === maxRetries) {
+        break;
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, delay));
+      delay = Math.min(delay * backoffFactor, maxDelay); // Cap the delay
+      retries++;
+    }

```

Comment:
Nope. Imagine maxRetries == 1. Then function will try this request once, than REtry it exactly 1 time. If one want to disable retries completely, they would just set maxRetries to 0. If it would lead to not trying requests at all, I would call it flaw in the logic